lib/tcr/cassette.rb

```diff
@@ -1,22 +1,23 @@
+require 'fileutils'
+
 module TCR
   class Cassette
     attr_reader :name
 
     def initialize(name)
       @name = name
+      @sessions = recording? ? [] : unmarshal(File.read(filename))
 
-      if File.exists?(filename)
-        @recording = false
-        @contents = File.open(filename) { |f| f.read }
-        @sessions = unmarshal(@contents)
-      else
-        @recording = true
+      if recording?
+        verify_cassette_path_is_writable
         @sessions = []
+      else
+        @sessions = unmarshal(File.read(filename))
```

Comment:
I don't quite follow the flow above.

First you use the ternary to set sessions to blank "if `recording?` .. then you if/else it again, which sets sessions to empty array again.. or unmarshal's it again.

Can you you help me understand?

And `recording?` seems inverted.. since you added `!File.exists`

Reply:
Whoops, 100% mistake :sweat:.

1. Pushed a fix last week, but forgot to follow up on this thread.

2. > And `recording?` seems inverted.. since you added `!File.exists`

   The original stanza was

   ```ruby
   if File.exists?(filename)
     @recording = false
     @contents = File.open(filename) { |f| f.read }
     @sessions = unmarshal(@contents)
   else
     @recording = true
     @sessions = []
   end
   ```

   Thus, if the file doesn't exist, we're `@recording` and `@sessions` is set to an empty array. The logic looks identical to me (but maybe I'm missing something?).

3. One catch that I just noticed is with the `||=` operator in my version of the `#recording?` method:

   ```ruby
   def recording?
     @recording ||= !File.exists?(filename)
   end
   ```

   If the file _does_ exist, this will call `File.exists?` every time it's called, since a `false` value for `@recording` will cause the rest of the expression to be evaluated. However, I think that the performance cost will be entirely negligible, and the added readiability is worth it.